src/Diagnostics.RuntimeHost/Program.cs

```diff
@@ -55,6 +55,10 @@ public static IHostBuilder CreateHostBuilder(string[] args)
                 webbuilder.ConfigureAppConfiguration((context, config) =>
                 {
                     var builtConfig = config.Build();
+                    config.AddEnvironmentVariables()
+                        .AddCommandLine(args)
+                        .Build();
+
```

Comment:
So by adding this earlier in the config we are saying that environment variables and command line arguments will hold lesser precedence than the appsettings captured in the encrypted json file, right?

Reply:
Oops. I did not realize that. I will revert this back. I moved it up to remove the confusion with `Secrets:Enabled` flag. Env variables & command line arguments should take more precedence than encrypted json.

Reply:
What is the confusion you are trying to clear with `Secrets:Enabled`? Maybe we can do a different change to help

Reply:
Remember we used to replace the `Secrets:Enabled `flag to false when deploying to prod endpoints? I thought moving this up will make the env variables take precedence and not go to keyvault to fetch secrets in the next line. But we have encrypted settings that comes after that, which I missed